src/adapter/src/coord/introspection.rs

```diff
@@ -86,7 +90,6 @@ pub fn auto_run_on_introspection<'a, 's, 'p>(
         | Plan::EmptyQuery
         | Plan::ShowAllVariables
         | Plan::ShowCreate(_)
-        | Plan::ShowColumns(_)
         | Plan::ShowVariable(_)
         | Plan::InspectShard(_)
         | Plan::SetVariable(_)
```

Comment:
I wonder if we should completely remove the early return here and just yield something like `(BTreeSet::empty(), false)`.

Reply:
Good thought! The tricky part is we'd actually want to return something like `(set[all], true)` because if there are no dependencies and not an expensive function, then we would route to `mz_introspection`, which isn't what we want by default. We could return something like `(BTreeSet::empty(), true)`, but we're not necessarily running an expensive function. I think for now I'll keep it as-is and we can revisit if we hit it again